src/libfetchers/fetchers.cc

```diff
@@ -336,6 +337,14 @@ std::pair<ref<SourceAccessor>, Input> Input::getAccessorUnchecked(ref<Store> sto
 
             accessor->fingerprint = getFingerprint(store);
 
+            // NOTE: This avoids https://github.com/NixOS/nix/issues/11228!
+            // See https://github.com/NixOS/nix/pull/12911 & https://github.com/NixOS/nix/pull/14041
```

Comment:
```suggestion            
            // Store a cache entry for the substituted tree so later fetches
            // can reuse the existing nar instead of copying the unpacked
            // input back into the store on every evaluation.

```